test/common/test_runner.go

```diff
@@ -133,6 +135,55 @@ func CreateNamespaceIfNeeded(t *testing.T, client *Client, namespace string) {
 		if err != nil {
 			t.Fatalf("The default ServiceAccount was not created for the Namespace: %s", namespace)
 		}
+
+		// If the "default" Namespace has a secret called
+		// "kn-eventing-test-pull-secret" then use that as the ImagePullSecret
+		// on the "default" ServiceAccount in this new Namespace.
+		// This is needed for cases where the images are in a private registry.
+
+		// Get the Interfaces we need to access the resources in the cluster
+		defSecI := client.Kube.Kube.CoreV1().Secrets("default")
+		nsSAI := client.Kube.Kube.CoreV1().ServiceAccounts(namespace)
+		nsSecI := client.Kube.Kube.CoreV1().Secrets(namespace)
+
+		testSecret, _ := defSecI.Get(TestPullSecretName, metav1.GetOptions{})
+
+		// Check again. I've seen cases where it lies and if we need it
```

Comment:
this seems odd, and is there a reason why you don't check the err returned? I'm not sure what you mean by "It" lies, are you saying above line returns nil, but the one below returns true?

Reply:
No I mean that the apiserver will return nil+err even though the secret does exist. I can't explain why or how, but I ran into it often enough that I decided to just ask again to be sure.

Reply:
@vagababov Have you seen something like this before? Seems b0rk3n?

Reply:
Not really. What _is_ the actual error you're getting? If I were to guess you got either network problem error (connection refused, dial timeout) or overload error from API server.

Which leads me to the main problem I see here: "never ignore errors".

Also, I'd highly recommend to switch to informers.

Reply:
What's the err that's returned?

Reply:
Let me change the code to show the error (if I even get one) to see what's up..  But to add more intrigue, I've also seen cases like this....

I had this code:
```
		if testSecret != nil {
			// Found the secret, so now make a copy in our new namespace
			newSecret, err := nsSecI.Create(
				&corev1.Secret{
					ObjectMeta: metav1.ObjectMeta{
						Name: testSecret.ObjectMeta.Name,
					},
					Data: testSecret.Data,
					Type: testSecret.Type,
				})
			if err != nil {
				t.Fatalf("TestSetup: Error copying the secret: %s", err)
			}
```
Notice I would use the "name" from "testSecret", which should never be empty. Yet I would get this error:
```
test_runner.go:82: namespace is : "test-default-broker-with-many-deprecated-triggers" test_runner.go:168: 
TestSetup: Error copying the secret: Secret "" is invalid: metadata.name: Required value: name or
generateName is required
```
I never saw this on my local testing, or IKS, only using Prow.